src/ActionBranchToPaletteCell.ts

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2023 Inclusive Design Research Centre, OCAD University
+ * All rights reserved.
+ *
+ * Licensed under the New BSD license. You may not use this file except in
+ * compliance with this License.
+ *
+ * You may obtain a copy of the License at
+ * https://github.com/inclusive-design/adaptive-palette/blob/main/LICENSE
+ */
+
+import { render } from "preact";
+import { html } from "htm/preact";
+import { BlissSymbolCellType } from "./index.d";
+import { adaptivePaletteGlobals, loadPaletteFromJsonFile } from "./GlobalData";
+import { Palette } from "./Palette";
+import { BlissSymbol } from "./BlissSymbol";
+import { speak } from "./GlobalUtils";
+import "./ActionBranchToPaletteCell.scss";
+
+// TODO:  this is identical to `ActionBmwCodeCellPropsType`.  Should it be?
+type ActionBranchToPalettePropsType = {
```

Comment:
I agree, and will start work on it next.  I also noticed that we don't define the return type of many of our functions.  Should we?

Reply:
It's a great idea to define the return type for functions.

Reply:
I've done them all.  The one that was problematic was the return type for the `htm()` template function used by all of the component Functions.  I thought `htm` returned a `string` type, but it does not.  Its return type is something like `TemplateStringsArray`, but when I looked into the project's repository, I found issue #73, [Question: How well does this work with TypeScript?](https://github.com/developit/htm/issues/73).  The main answer is given in the [first response](https://github.com/developit/htm/issues/73#issuecomment-468870790): "there aren't typings for htm / html, but there are extremely basic types for the react and preact integrations."  The rest of the discussion is about TypeScript's effort to support a tagged template type.  That work is ongoing.

I did some more debugging and it looks like, in the case of Preact, the `htm` return type is `VNode`, or "virtual DOM node".  I'm currently using that for the return type.

As for the expanding the options, I'm still working on that.

Reply:
> I'm thinking if we should expand the `options` section for every type component as this part is not defined and verified by typescript. What do you think?

For "expanding the options section", I took the full `BlissSymbolCellType` and created new types from its parts.  Since these are not palette cell types, I used "Info" in their type names; for example, a `LayoutInfoType` that contains only the grid-styles information.  Then, one can define cell types by combining the info types as needed.

Let me know what you think.